packages/itwin/tree-widget/src/tree-widget-react/components/trees/imodel-content-tree/IModelContentTree.tsx

```diff
@@ -3,44 +3,55 @@
  * See LICENSE.md in the project root for license terms and full copyright notice.
  *--------------------------------------------------------------------------------------------*/
 
+import { useCallback, useMemo } from "react";
 import { SvgFolder, SvgGroup, SvgHierarchyTree, SvgImodelHollow, SvgItem, SvgLayers, SvgModel } from "@itwin/itwinui-icons-react";
 import { Tree } from "../common/components/Tree.js";
 import { TreeRenderer } from "../common/components/TreeRenderer.js";
 import { IModelContentTreeComponent } from "./IModelContentTreeComponent.js";
-import { IModelContentTreeDefinition } from "./IModelContentTreeDefinition.js";
+import { defaultHierarchyConfiguration, IModelContentTreeDefinition } from "./IModelContentTreeDefinition.js";
 import { IModelContentTreeIdsCache } from "./internal/IModelContentTreeIdsCache.js";
 
-import type { TreeProps } from "../common/components/Tree.js";
 import type { ReactElement } from "react";
+import type { IModelContentTreeHierarchyConfiguration } from "./IModelContentTreeDefinition.js";
+import type { TreeProps } from "../common/components/Tree.js";
 import type { PresentationHierarchyNode } from "@itwin/presentation-hierarchies-react";
 
 /** @beta */
 export type IModelContentTreeProps = Pick<TreeProps, "imodel" | "getSchemaContext" | "selectionStorage" | "density" | "selectionMode"> & {
   hierarchyLevelConfig?: {
     sizeLimit?: number;
   };
+  hierarchyConfig?: Partial<IModelContentTreeHierarchyConfiguration>;
 };
 
 /** @beta */
-export function IModelContentTree(props: IModelContentTreeProps) {
+export function IModelContentTree({ hierarchyConfig, ...props }: IModelContentTreeProps) {
+  const hierarchyConfiguration = useMemo<IModelContentTreeHierarchyConfiguration>(
+    () => ({
+      ...defaultHierarchyConfiguration,
+      ...hierarchyConfig,
+    }),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    Object.values(hierarchyConfig ?? {}),
+  );
```

Comment:
Looks like this could be moved into `useCallback` below.

Reply:
agree, moved